src/actions/vstudio/vs2010_vcxproj.lua

```diff
@@ -1231,7 +1231,15 @@
 
 
 	function m.generateDebugInformation(cfg)
-		m.element("GenerateDebugInformation", nil, tostring(cfg.flags.Symbols ~= nil))
+		if cfg.flags.Symbols and _ACTION >= "vs2015" then
+			if cfg.flags.FastLink then
+				m.element('GenerateDebugInformation', nil, "DebugFastLink")
+			else
+				m.element('GenerateDebugInformation', nil, "Debug")
```

Comment:
It's a bummer that OS, toolset and action agnostic is somewhat of a pipedream... this is not a problem of premake really, CMake and many other systems have similar problems... all those build systems are just too damn different.

Anyway, I'm for proper API... I'll see if I can do that.... "symbolinformation" seems like a proper name no?
And then valid values are: "Default", "On", "Off", "FastLink", where "Default" will not emit the element, "On" is explicitly on, off explicitly off, and FastLink==On on all platforms but VS2015.